sympy/matrices/tests/test_determinant.py

```diff
@@ -12,14 +12,27 @@ class DeterminantOnlyMatrix(_MinimalMatrix, _CastableMatrix, MatrixDeterminant):
     pass
 
 def test_determinant():
+    M = Matrix()
 
-    for M in [Matrix(), Matrix([[1]])]:
-        assert (
-            M.det() ==
-            M._eval_det_bareiss() ==
-            M._eval_det_berkowitz() ==
-            M._eval_det_lu() ==
-            1)
+    assert M.det() == 1
+    # Evaluating these directly because they are never reached via M.det()
```

Comment:
We could use `M.det(method="bareiss")` format consistently, but previous code seemed to care about algorithms giving the right answer too :-)